docs/archetypes/authoring-guide.md

```diff
@@ -333,32 +333,110 @@ module telemetryCustomerUsageAttribution '../../azresources/telemetry/customer-u
+
+The ARM parameter file name can be in one of two formats:
+
+- `[SubscriptionGUID]`\_`[ArchetypeName]`.json
+- `[SubscriptionGUID]`\_`[ArchetypeName]`\_`[DeploymentLocation]`.json
+
+The subscription GUID is required by the pipeline to target the archetype deployment.
+
+The template name/type is a text fragment corresponding to a path name (or part of a path name) under the '/landingzones' top level path. It indicates which Bicep templates to run on the subscription. For example, the machine learning path is `/landingzones/lz-machinelearning`, so we remove the `lz-` prefix and use `machinelearning` to specify this type of landing zone (archetype).
 
-The JSON config file name is in one of the following two formats:
+The deployment location is the short name of an Azure deployment location, which may be used to override the `deploymentRegion` YAML variable.  This parameter is configurable in `common.yml` and you can also override in environment configuration files.  By default it is set to `canadacentral`.  The allowable values for this value can be determined by looking at the `Name` column output of the command: `az account list-locations -o table`.
 
-- [AzureSubscriptionGUID]\_[TemplateName].json
-- [AzureSubscriptionGUID]\_[TemplateName]\_[DeploymentLocation].json
+### Management Group Id detection
+
+Management Group Id must be unique within an Azure environment.  Since a management group hierarchy can be up to 6 levels deep with many child management groups , it is important to ensure uniqueness across all management groups ids.
+
+There are two approaches for achieving uniquness:
+
+1. Use the parent management group id as the prefix for every child management group id.  For example:
+
+    ```none
+        - pubsec
+          - pubsecLandingZones
+            - pubsecLandingZonesDevTest
+    ```
+
+    In this approach, the prefix creates the natural uniqueness.
+
+2. Create a management group hierarchy where every management group id can be unique and it is manually checked.  For example:
+
+    ```none
+        - pubsec
+          - LandingZones
+            - DevTest
+    ```
 
-The subscription GUID is needed by the pipeline; since it's not available in the file contents, it is specified in the config file name.
+    In this approach, you must ensure all management group ids are unique yourself.
 
-The template name/type is a text fragment corresponding to a path name (or part of a path name) under the '/landingzones' top level path. It indicates which Bicep templates to run on the subscription. For example, the machine learning path is `/landingzones/lz-machinelearning`, so we remove the `lz-` prefix and use `machinelearning` to specify this type of landing zone.
+The `subscription-ci` management group detection logic is built based on approach #1.  However it will also work for both scenarios.
```

Comment:
Correction: "the `subscription-ci` management group detection logic was/is built to accommodate both scenarios" (plus many others that we don't describe here, e.g. `pubsec/L/a/n/d/i/n/g/Z/o/n/e/s/DevTest`.